src/realm/object-store/sync/sync_session.cpp

```diff
@@ -244,21 +268,22 @@ void SyncSession::become_waiting_for_access_token()
     m_state = State::WaitingForAccessToken;
 }
 
-void SyncSession::handle_location_update_failed(Status status)
+void SyncSession::become_waiting_for_location()
 {
-    // TODO: ideally this would write to the logs as well in case users didn't set up their error handler.
-    {
-        util::CheckedUniqueLock lock(m_state_mutex);
-        // Should only be called while waiting to update the access token
-        REALM_ASSERT(m_state == State::WaitingForAccessToken);
-        // Close the session, since there's nothing more to do at this point, it can be resumed
-        // after resolving the location update failure
-        become_inactive(std::move(lock), status);
-    }
-    if (auto error_handler = config(&SyncConfig::error_handler)) {
-        auto user_facing_error = SyncError({ErrorCodes::SyncConnectFailed, status.reason()}, true);
-        error_handler(shared_from_this(), std::move(user_facing_error));
+    REALM_ASSERT(m_state != State::WaitingForLocation);
+    m_state = State::WaitingForLocation;
+    m_sync_manager->start_update_location();
+}
+
+void SyncSession::handle_location_updated()
+{
+    util::CheckedUniqueLock lock(m_state_mutex);
+    // If the state is not waiting for location, bail early
+    if (m_state != State::WaitingForLocation) {
+        return;
```

Comment:
Since we're going through the list of SyncSessions (with external references) to notify them of the location update, we wouldn't want to restart any sessions that are currently paused.